packages/template-retail-react-app/app/components/search/index.test.js

```diff
@@ -125,3 +151,261 @@ test('passing undefined to Suggestions returns undefined', async () => {
+})
+
+test('when sendTextMessage fails and launchChat returns maximized message, no additional send text is triggered', async () => {
+    const user = userEvent.setup()
+
+    getConfig.mockImplementation(() =>
+        getMockedConfigWithCommerceAgentSettings(mockConfig, 'true', 'true')
+    )
+
+    // Create spies for chat functions
+    const sendTextMessageSpy = jest
+        .fn()
+        .mockRejectedValue(
+            'invoke API before the onEmbeddedMessagingConversationOpened event is fired'
+        )
+    const launchChatSpy = jest.fn().mockResolvedValue('Successfully maximized the messaging client')
+
+    // Mock window.embeddedservice_bootstrap
+    window.embeddedservice_bootstrap = {
+        utilAPI: {
+            sendTextMessage: sendTextMessageSpy,
+            launchChat: launchChatSpy
+        }
+    }
+
+    renderWithProviders(<SearchInput />)
+    const searchInput = document.querySelector('input[type="search"]')
+
+    // Perform a search
+    await user.type(searchInput, 'test search{enter}')
+
+    // Wait for the setTimeout in onSubmitSearch
+    await new Promise((resolve) => setTimeout(resolve, 500))
+
+    // Verify sendTextMessage was called and failed
+    expect(sendTextMessageSpy).toHaveBeenCalledWith('test search')
+    expect(launchChatSpy).toHaveBeenCalled()
+
+    // Wait for any potential setTimeout after launchChat
+    await new Promise((resolve) => setTimeout(resolve, 500))
+
+    // Verify sendTextMessage was only called once
+    expect(sendTextMessageSpy).toHaveBeenCalledTimes(1)
+})
+
+test('when sendTextMessage and launchChat both fail, no additional send text is triggered', async () => {
+    const user = userEvent.setup()
+
+    getConfig.mockImplementation(() =>
+        getMockedConfigWithCommerceAgentSettings(mockConfig, 'true', 'true')
+    )
+
+    // Create spies for chat functions
+    const sendTextMessageSpy = jest
+        .fn()
+        .mockRejectedValue(
+            'invoke API before the onEmbeddedMessagingConversationOpened event is fired'
+        )
+    const launchChatSpy = jest.fn().mockRejectedValue('Failed to launch chat')
+
+    // Mock window.embeddedservice_bootstrap
+    window.embeddedservice_bootstrap = {
+        utilAPI: {
+            sendTextMessage: sendTextMessageSpy,
+            launchChat: launchChatSpy
+        }
+    }
+
+    renderWithProviders(<SearchInput />)
+    const searchInput = document.querySelector('input[type="search"]')
+
+    // Perform a search
+    await user.type(searchInput, 'test search{enter}')
+
+    // Wait for the setTimeout in onSubmitSearch
+    await new Promise((resolve) => setTimeout(resolve, 500))
+
+    // Verify sendTextMessage was called and failed
+    expect(sendTextMessageSpy).toHaveBeenCalledWith('test search')
+    expect(launchChatSpy).toHaveBeenCalled()
+
+    // Wait for any potential setTimeout after launchChat
+    await new Promise((resolve) => setTimeout(resolve, 500))
```

Comment:
@sf-mkosak nitpick -- you could use `jest.advanceTimersByTime(ms)` to prevent a scenario where this test adds unneeded latency to the (already fairly slow) test runs